AwesomeVideoPlayer/AwesomeVideoPlayer/Controllers/VideoPlayViewController.swift

```diff
@@ -24,20 +49,60 @@ class VideoPlayViewController: UIViewController {
             return
         }
         playerLayer.player = playerView
-        playerLayer.videoGravity = .resizeAspect
-        playerView.play()
+        
+        playerLayer.videoGravity = .resizeAspect // CMTime interval = CMTimeMakeWithSeconds(1.0, NSEC_PER_SEC);
+        self.timeObserverToken = playerView.addPeriodicTimeObserver(forInterval: CMTimeMakeWithSeconds(1, preferredTimescale: 1), queue: DispatchQueue.main) { (time) in
+            let currentTime = CMTimeGetSeconds(time)
+            let duration = Float(self.networkVideo.videoDuration / 1000)
+            self.timeLabel.text = "\(currentTime.toString()) | \(self.durationString)"
+            self.durationSlider.value = Float(Float(currentTime) / duration)
+        }
     }
     
-    init(withVideo video: NetworkVideo) {
-        guard let videoURL = URL(string: video.videoURL) else {
-            fatalError("Failed to load video URL: \(video.videoURL)")
+    override func viewWillDisappear(_ animated: Bool) {
+        if let observer = timeObserverToken {
+            playerView.removeTimeObserver(observer)
         }
-        playerView = AVPlayer(url: videoURL)
-        super.init(nibName: "VideoPlayViewController", bundle: nil)
     }
     
-    required init?(coder aDecoder: NSCoder) {
-        fatalError("init coder aDecoder: not implemented")
+    @IBAction func closeTapped(_ sender: UIButton) {
+        playerView.pause()
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func userBeganScroll(_ sender: UISlider) {
+        playerView.pause()
+        playButton.isSelected = false
     }
+    @IBAction func userFinishedScroll(_ sender: UISlider) {
+        guard let currentTime = playerView.currentItem?.asset.duration else {
+            return
+        }
+        let newCurrentTime = Double(sender.value) * CMTimeGetSeconds(currentTime)
+        let seekTime = CMTimeMakeWithSeconds(newCurrentTime, preferredTimescale: 600)
+        playerView.seek(to: seekTime)
+    }
+    @IBAction func userScrolling(_ sender: UISlider) {
+        guard let duration = playerView.currentItem?.asset.duration else {
+            return
+        }
+        let seconds : Float64 = CMTimeGetSeconds(duration) * Double(sender.value)
+        self.timeLabel.text = "\(seconds.toString()) | \(self.durationString)"
+    }
+    
+}
 
+extension VideoPlayViewController {
+    /// Toggle play or pause for video
+    @IBAction func playToggled(_ sender: UIButton) {
```

Comment:
I think I should move all IBAction code to the extension so I can slowly remove all the view code from VC